### PR TITLE
Run cargo fmt on the whole project

### PR DESCRIPTION
**Pull request by CohenArthur**

### CODE REVIEW

File: src/abi/mod.rs [hunk 1]
```diff
@@ -14,7 +14,10 @@ use crate::prelude::*;
 pub(crate) use self::returning::{can_return_to_ssa_var, codegen_return};
 
 // Copied from https://github.com/rust-lang/rust/blob/f52c72948aa1dd718cc1f168d21c91c584c0a662/src/librustc_middle/ty/layout.rs#L2301
-pub(crate) fn fn_sig_for_fn_abi<'tcx>(tcx: TyCtxt<'tcx>, instance: Instance<'tcx>) -> ty::PolyFnSig<'tcx> {
+pub(crate) fn fn_sig_for_fn_abi<'tcx>(
```
Comment (bjorn3) [inline src/abi/mod.rs:17]: Please use `#[rustfmt::skip]` on this function. It is supposed to be a 1-to-1 copy of the rustc version. While I have to slightly change to to work with the different imports in this file, I would like to minimize the diff when I have to sync it with upstream again.

File: src/target_features_whitelist.rs [hunk 2]
```diff
@@ -107,8 +107,10 @@ const POWERPC_ALLOWED_FEATURES: &[(&str, Option<Symbol>)] = &[
     ("vsx", Some(sym::powerpc_target_feature)),
 ];
 
-const MIPS_ALLOWED_FEATURES: &[(&str, Option<Symbol>)] =
```
Comment (bjorn3) [inline src/target_features_whitelist.rs:110]: Please add `#![rustfmt::skip]` at the top of this file. It is a literal copy a big chunk of the equivalent rustc file.

File: src/optimize/mod.rs [hunk 3]
```diff
@@ -5,8 +5,7 @@ mod stack2reg;
 
 pub(crate) fn optimize_function<'tcx>(
     tcx: TyCtxt<'tcx>,
-    #[cfg_attr(not(debug_assertions), allow(unused_variables))]
-    instance: Instance<'tcx>,
+    #[cfg_attr(not(debug_assertions), allow(unused_variables))] instance: Instance<'tcx>,
```
Comment (bjorn3) [inline src/optimize/mod.rs:8]:
```suggestion
    instance: Instance<'tcx>,
```

It has been used in all cases for a while.

File: src/abi/mod.rs (hunk 1 above)
Comment (CohenArthur) [reply src/abi/mod.rs]: Fixed in 9c848e1

File: src/target_features_whitelist.rs (hunk 2 above)
Comment (CohenArthur) [reply src/target_features_whitelist.rs]: Fixed in ed35bc8

File: src/target_features_whitelist.rs [hunk 4]
```diff
@@ -1,3 +1,5 @@
+#[rustfmt::skip]
```
Comment (bjorn3) [inline src/target_features_whitelist.rs:1]:
```suggestion
#![rustfmt::skip]
```

Comment (CohenArthur) [reply src/target_features_whitelist.rs:1]: Oops, good catch

**Review by bjorn3 — APPROVED** [review]
Thanks!

**Comment by bjorn3** [comment]
Forgot to merge.

**Pull request merged**